app/Models/Internet/InternetAccess.php

```diff
@@ -2,45 +2,98 @@
 
 namespace App\Models\Internet;
 
-use Illuminate\Database\Eloquent\Model;
+use App\Models\Semester;
 use App\Models\User;
+use Eloquent;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\Eloquent\Model;
 use Illuminate\Database\Eloquent\Relations\BelongsTo;
 use Illuminate\Database\Eloquent\Relations\HasMany;
+use Illuminate\Support\Carbon;
 use Illuminate\Support\Str;
 
 /**
+ * App\Models\Internet\InternetAccess
+ *
+ * The InternetAccess model connects the user to the internet related models.
+ * For Wi-Fi connections,
+ *
  * @property mixed $user_id
  * @property string $wifi_username
  * @property Carbon $has_internet_until
  * @property string $wifi_password
- * @property int $auto_approved_mac_slots
  * @property User $user
  * @property WifiConnection[]|Collection $wifiConnections
  * @method $isActive
  * @method $setWifiCredentials
  * @method $resetPassword
+ * @property Carbon|null $created_at
+ * @property Carbon|null $updated_at
+ * @property-read int|null $wifi_connections_count
+ * @method static Builder|InternetAccess newModelQuery()
+ * @method static Builder|InternetAccess newQuery()
+ * @method static Builder|InternetAccess query()
+ * @method static Builder|InternetAccess whereAutoApprovedMacSlots($value)
+ * @method static Builder|InternetAccess whereCreatedAt($value)
+ * @method static Builder|InternetAccess whereHasInternetUntil($value)
+ * @method static Builder|InternetAccess whereUpdatedAt($value)
+ * @method static Builder|InternetAccess whereUserId($value)
+ * @method static Builder|InternetAccess whereWifiPassword($value)
+ * @method static Builder|InternetAccess whereWifiUsername($value)
+ * @property-read Collection|MacAddress[] $macAddresses
+ * @property-read int|null $mac_addresses_count
+ * @mixin Eloquent
```

Comment:
These IDE helpers would have been nice in a separate PR :D

Reply:
well yes, sorry